### DISCUSSION TITLE
Error restoring backup in WhatsApp

### DISCUSSION THREAD

**Discussion by akshayi1** [ElDavoo/wa-crypt-tools#154] [General]
Hello! I realize this may not be the best place, but after reading multiple discussions, I figured that this may be the place where I can at least get some level of certainty around if I can ever restore my messages or not. The following is from my post on r/whatsapp, but still holds good. I'd thoroughly appreciate any insight or help of any kind. Thank you in advance!

I've been using WhatsApp for over 10 years and outside of a few times over 8 years ago, I've always safely been able to transfer chats across phones and SIMs and it's worked like a charm every single time. Except this time. 

Some context: I was on the beta version, which last week, when updated, began crashing whenever I'd open the app, making is unusable. After waiting for more than a day for a fix to arrive, I decided to leave the beta, uninstall WhatsApp and reinstall it. I think this is where I screwed up, but I can't be too sure. Here's what happened next.

1. When I reinstalled it, rather than directly get the backup from Google Drive and restore from there, it first asked for my encryption password (or the 64-bit key, if I had that instead). I found this odd, because I had not set up E2E encryption to begin with, and so didn't know how to proceed, and so exit the process. I tried a few passwords, which didn't work before exiting though.
2. Did some reading up and saw that the only way to reset the password would be to set a new one. So I uninstalled, reinstalled, skipped the restore and then set a new password.
3. Armed with a new password, I then uninstalled, reinstalled and used this new password, at which point, it went about restoring the backup from Google Drive. A few minutes later, I got an error - "Sorry, we were unable to restore any of your message history backups." and it hadn't restored any of my messages (over 8 years worth) and instead began restoring my media (which I don't really care much about).

I have the db.crypt15 files - one big 110 MB one and a few incremental files (which I'd ideally merge into one, but I'd be just as happy with the largest file to begin with), but WhatsApp refuses to restore any of the messages from them. 

One of the suggestions was to, after the initial set up screen, put the phone in Airplane Mode and force it to restore from the local, but when I do, after putting the new password (and it validates it too) I get the same error message as in step 3.

I'm almost ready to give up and use this as a fresh start to trim my messages and contacts, but at the same time, I want to try everything I can to get this back.

Does anyone know how I can do this? Again, I do NOT have the 64-bit key and only the password. If there's a way to restore the backup, and setup a new key which can then also be applied to previously created files, AND then that can be used again to restore from there, please let me know. Thank you!

Extras:

1. I reached out to WhatsApp support and they said there is a bug with backups that they are investigating. 
2. When I let the restoration to go through without the chats, since it downloads the chats, I can see a new msgstore.db.crypt file being created. Is this something that's possible - I let it do it's thing without restoring the chats, reset the password and use the 64-bit key this time, uninstall and reinstall WhatsApp (again), point it to this newly created msgstore.db.crypt file, and use the 64-bit key to successfully restore the chats? 

In my head, it makes sense, but nothing I've tried over the last week that logically makes sense actually works. Please help. Thank you!

Even more extras:

I just used wainfo on the large 110 MB file and got this output -

wainfo.py:50 : [W] This script is in beta stage.
keyfactory.py:31 : [D] Reading keyfile...
keyfactory.py:43 : [C] The keyfile is not a valid Java object: Invalid file magic: 0x8501
keyfactory.py:55 : [C] Unrecognized key file format.
None

Did I screw things up when I left the beta and installed the non-beta version of WhatsApp? :(

**Comment by ElDavoo** [comment]
> I think this is where I screwed up, but I can't be too sure.

Yep, databases can be upgraded, but not downgraded.

> Sorry, we were unable to restore any of your message history backups." and it hadn't restored any of my messages (over 8 years worth) and instead began restoring my media (which I don't really care much about).

Did you uninstall in the middle of a backup I wonder? Or the backup in google server was incomplete.

> after putting the new password (and it validates it too) I get the same error message as in step 3.

See [here](https://github.com/ElDavoo/wa-crypt-tools#can-i-use-the-password-to-decrypt-the-database) . The password is sent to WA server, and if it's correct, you receive the 256-bit key.

> I'm almost ready to give up and use this as a fresh start to trim my messages and contacts

Yup, decluttering is a good practice. After storage space, things also take mental space.


If support said there is a bug, i'd just wait, maybe newer betas will fix.

Anyway, I don't know anything about how WA handle keys, partial uploads, restoring, etc, I just made a program to handle the file formats.

**Comment by osmaircoelho** [comment]
I have the same issue @akshayi1

Phone: Motorola edge 50 pro
Android version: 15
Wapp version: 2.24.25.77

I'm trying to restore it. I've tried every way, but it always gives an error at the end and I always get the message: "Sorry, we are unable to restore any of your message history backup."

I only have 399MB of messages and 4.4GB of media, which is not much, but it doesn't work.

I've searched all the forums, but there are a lot of people who are having the same problem. I have 10+ messages that could be lost. I've already deleted the cloud backup and redid it on my old phone.

I'm waiting for WA support.

**Reply by silencoder** [reply]
Any update on this?

**Comment by ElDavoo** [comment]
I think you shouldn't do this, as you cannot downgrade the database.
So if this happens to you, you would have to wait until the stable version becomes newer than your version.
